src/hotspot/share/opto/parseHelper.cpp

```diff
@@ -523,7 +523,7 @@ void PEAState::add_new_allocation(GraphKit* kit, Node* obj) {
     }
 
     ciMethod* method = kit->jvms()->method();
-    if (PEAContext::instance().match(method)) {
+    if (PEAContext::instance().match(method) || kit->C->has_pea_method_only()) {
```

Comment:
'method' here is the current method, not the top-level method. that's why I can support inlined method. 

On the other side, your option gets set only one chance for each compilation unit.